File: generate-icons.js
const fs = require('fs');
const { createCanvas } = require('canvas');

const sizes = [72, 96, 128, 144, 152, 192, 384, 512];

sizes.forEach(size => {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');

    // Background
    ctx.fillStyle = '#00d4ff';
    ctx.fillRect(0, 0, size, size);

    // Inner square
    const padding = size * 0.1;
    ctx.fillStyle = '#000';
    ctx.fillRect(padding, padding, size - 2*padding, size - 2*padding);

    // Code brackets and lines
    ctx.strokeStyle = '#00d4ff';
    ctx.fillStyle = '#00d4ff';
    ctx.lineWidth = size * 0.03;
    ctx.lineCap = 'round';
    ctx.lineJoin = 'round';

    // Left bracket
    ctx.beginPath();
    ctx.moveTo(size * 0.27, size * 0.35);
    ctx.lineTo(size * 0.20, size * 0.50);
    ctx.lineTo(size * 0.27, size * 0.65);
    ctx.stroke();

    // Right bracket
    ctx.beginPath();
    ctx.moveTo(size * 0.73, size * 0.35);
    ctx.lineTo(size * 0.80, size * 0.50);
    ctx.lineTo(size * 0.73, size * 0.65);
    ctx.stroke();

    // Blur lines
    const lineHeight = size * 0.024;
    const radius = lineHeight / 2;

    // Line 1
    ctx.beginPath();
    ctx.roundRect(size * 0.35, size * 0.39, size * 0.16, lineHeight, radius);
    ctx.fill();

    // Line 2
    ctx.beginPath();
    ctx.roundRect(size * 0.35, size * 0.49, size * 0.30, lineHeight, radius);
    ctx.fill();

    // Line 3
    ctx.beginPath();
    ctx.roundRect(size * 0.35, size * 0.59, size * 0.23, lineHeight, radius);
    ctx.fill();

    // Save
    const buffer = canvas.toBuffer('image/png');
    fs.writeFileSync(`icons/icon-${size}x${size}.png`, buffer);
    console.log(`Generated icon-${size}x${size}.png`);
});

console.log('All icons generated!');

File: generate-icons.py
#!/usr/bin/env python3
from PIL import Image, ImageDraw
import os

# Create icons directory if it doesn't exist
os.makedirs('icons', exist_ok=True)

sizes = [72, 96, 128, 144, 152, 192, 384, 512]

for size in sizes:
    # Create image
    img = Image.new('RGB', (size, size), '#00d4ff')
    draw = ImageDraw.Draw(img)

    # Inner square
    padding = int(size * 0.1)
    draw.rectangle(
        [padding, padding, size - padding, size - padding],
        fill='#000000'
    )

    # Code brackets
    line_width = max(2, int(size * 0.03))

    # Left bracket
    left_x = int(size * 0.20)
    left_outer_x = int(size * 0.27)
    top_y = int(size * 0.35)
    middle_y = int(size * 0.50)
    bottom_y = int(size * 0.65)

    draw.line([(left_outer_x, top_y), (left_x, middle_y)], fill='#00d4ff', width=line_width)
    draw.line([(left_x, middle_y), (left_outer_x, bottom_y)], fill='#00d4ff', width=line_width)

    # Right bracket
    right_x = int(size * 0.80)
    right_outer_x = int(size * 0.73)

    draw.line([(right_outer_x, top_y), (right_x, middle_y)], fill='#00d4ff', width=line_width)
    draw.line([(right_x, middle_y), (right_outer_x, bottom_y)], fill='#00d4ff', width=line_width)

    # Blur lines
    line_height = max(2, int(size * 0.024))
    line1_y = int(size * 0.39)
    line2_y = int(size * 0.49)
    line3_y = int(size * 0.59)
    start_x = int(size * 0.35)

    draw.rounded_rectangle(
        [start_x, line1_y, start_x + int(size * 0.16), line1_y + line_height],
        radius=line_height//2,
        fill='#00d4ff'
    )
    draw.rounded_rectangle(
        [start_x, line2_y, start_x + int(size * 0.30), line2_y + line_height],
        radius=line_height//2,
        fill='#00d4ff'
    )
    draw.rounded_rectangle(
        [start_x, line3_y, start_x + int(size * 0.23), line3_y + line_height],
        radius=line_height//2,
        fill='#00d4ff'
    )

    # Save
    filename = f'icons/icon-{size}x{size}.png'
    img.save(filename, 'PNG')
    print(f'Generated {filename}')

print('All icons generated!')
